Update the article

mkorobeinikov opened the pull request:
Update, fix, and impruve the structure of the article.

dmbaturin reviewed (changes requested):
Could you please use a descriptive commit message, like "Update the _Terraform_ article"? Ideally there should also be a brief summary of the update as well.

mkorobeinikov commented:
> Could you please use a descriptive commit message, like "Update the _Terraform_ article"? Ideally there should also be a brief summary of the update as well.

Accepted.
Update to the article about terraform. I made changes to the structure and divided the article into three parts AWS AZ vSPhere. Made changes to the table of contents of the article. Added links to resources in git. Added troubleshooting as well as a detailed description of the files.

dmbaturin commented:
Better than nothing but, frankly, still needs a lot of improvement. Let's just go with it for now, but I suggest that you look at commit messages of the Linux kernel, for example.
